server/src/editor.js

```diff
@@ -0,0 +1,202 @@
+        <button id="pen" onClick={this.onClickPen.bind(this)}>Pen</button>
+        <button id="text" onClick={this.onClickRectangle.bind(this)}>Rectangle</button>
+        <button id="arrow" onClick={this.onClickArrow.bind(this)}>Arrows</button>
+        <button id="erase" onClick={this.onClickErase.bind(this)}>Erase</button>
+        <Picker onSelectColor={this.handleColorSelect.bind(this)} onSelectSize={this.handleSizeSelect.bind(this)}></Picker>
+        <button id="save" onClick={this.onClickSave.bind(this)}>Save</button>
+        <button id="cancel" onClick={this.onClickCancel.bind(this)}>Cancel</button>
+        <button id="clear" onClick={this.onClickClear.bind(this)}>Clear</button>
+      </div>
+      <div className="main-container">
+        <img id="image" src={this.props.clip.image.url} style={{height: "auto", width: this.props.clip.image.dimensions.x + "px", maxWidth: "80%"}}/>
+        <div className="canvas-container" id="canvas-container">
+          <canvas className="image-holder" id="image-holder" ref="image" height={ 2 * canvasHeight } width={ 2 * window.innerWidth } style={{height: canvasHeight, width: window.innerWidth}}></canvas>
+          <canvas className="highlighter" id="highlighter" ref="highlighter" height={canvasHeight} width={window.innerWidth}></canvas>
+          <canvas className="editor" id="editor" ref="editor" height={canvasHeight} width={window.innerWidth}></canvas>
+        </div>
+      </div>
+    </div>
+  }
+
+  handleColorSelect(color) {
+    this.setState({color})
+  }
+
+  handleSizeSelect(size) {
+    this.setState({size});
+  }
+
+  componentDidUpdate() {
+    this.edit();
+  }
+
+  onClickErase() {
+    this.setState({tool: 'eraser'});
+  }
+
+  onClickCancel() {
+    this.props.onCancelEdit(false);
+  }
+
+  onClickSave() {
+    this.imageContext.drawImage(editor, 0, 0);
+    this.imageContext.globalCompositeOperation = 'multiply';
+    this.imageContext.drawImage(highlighter, 0, 0);
+    editor.style.display = 'none';
+    highlighter.style.display = 'none';
+    let url = this.imageCanvas.toDataURL("image/png");
+    window.open(this.imageCanvas.toDataURL("image/png"));
+  }
+
+  onClickClear() {
+    context.clearRect(0, 0, editor.width, editor.height);
+    highlightContext.clearRect(0, 0, highlighter.width, highlighter.height);
+  }
+
+  onClickHighlight() {
+    this.setState({tool: 'highlighter'});
+  }
+
+  onClickPen() {
+    this.setState({tool: 'pen'});
+  }
+
+  onClickArrow() {
+    this.setState({tool: 'arrow'});
+  }
+
+  onClickRectangle() {
+    this.setState({tool: 'rectangle'});
+  }
+
+  componentDidMount() {
+    editor = document.getElementById("editor");
+    context = this.refs.editor.getContext('2d');
+    highlighter = document.getElementById("highlighter");
+    highlightContext = this.refs.highlighter.getContext('2d');
+    let imageCanvas = document.getElementById("image-holder");
```

Comment:
There's a React way to get this node, maybe `this.ref.imageHolder` (and then `ref="imageHolder"`) on the element